plugins/janus_videoroom.c

```diff
@@ -1127,6 +1137,40 @@ char *janus_videoroom_query_session(janus_plugin_session *handle) {
 	return info_text;
 }
 
+static int janus_videoroom_access_room(json_t *root, gboolean check_secret, gboolean check_pin, janus_videoroom **videoroom) {
+	/* rooms_mutex has to be locked */
+	int error_code = 0;
+	char error_cause[512];
+	json_t *room = json_object_get(root, "room");
+	guint64 room_id = json_integer_value(room);
+	*videoroom = g_hash_table_lookup(rooms, GUINT_TO_POINTER(room_id));
+	if(*videoroom == NULL) {
+		JANUS_LOG(LOG_ERR, "No such room (%"SCNu64")\n", room_id);
+		error_code = JANUS_VIDEOROOM_ERROR_NO_SUCH_ROOM;
+		g_snprintf(error_cause, 512, "No such room (%"SCNu64")", room_id);
```

Comment:
My mistake, I need the contents of `error_cause` outside that function after an error.  I'll check in a new version with the additional parameters `error_cause` and `error_cause_size`.